fix(doc): Update executor plugins example (#13019)
argoproj/argo-workflows: docs/executor_plugins.md

      value: "true"
```

> In case you want to modify a running `workflow-controller`:

Reviewer: ```suggestion
If you want to modify a running Controller:
```

- simplify, "in case" -> "if", per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
- "controller" is how it is referred to in this doc and elsewhere, be consistent

        - containerPort: 4355
      securityContext:
        runAsNonRoot: true

>         runAsUser: 1000

Reviewer: can you elaborate why this is necessary? I'm not sure if this is necessary for all users, or just your specific environment

Author: Not sure if it's a local environmental problem, but leaving that to `65534` the examples never works. I provided more info in the original issue. I'll be happy to be proven wrong.

I use OS X (13.6 (22G120)) and Docker Desktop 4.14.1.

Reviewer: Yes you mentioned this in [the issue](https://github.com/argoproj/argo-workflows/issues/13015#issuecomment-2098378193) but you didn't say _why_ there either. That's why I asked here

Reviewer: This doesn't explain why it needs `runAsUser: 1000`, but apparently the tests do exactly that: https://github.com/argoproj/argo-workflows/blob/2f15709952075ba208cf45ec459e66a14fa37d02/test/e2e/manifests/plugins/hello-executor-plugin-configmap.yaml#L71 ... not sure why the docs were different 🤔

Reviewer: `git blame` shows it was actually added in #7855, not during the original feature #7256

Reviewer: I see it was added in [the docs too](https://github.com/argoproj/argo-workflows/blob/2f15709952075ba208cf45ec459e66a14fa37d02/docs/executor_plugins.md?plain=1#L250) in #7688.

I couldn't [find a place in the code](https://github.com/search?q=repo%3Aargoproj%2Fargo-workflows%20runAsUser&type=code) where this was actually enforced though.
The only relevant part seemed to be that the [default Executor `runAsUser` is `1000`](https://github.com/argoproj/argo-workflows/blob/2f15709952075ba208cf45ec459e66a14fa37d02/docs/workflow-controller-configmap.yaml#L189) (partly from #8062).

Author: Correct you can not enforce the `runAsUser` value for the `ExecutorPlugin` and you can customise that. I thought that was the whole point of documenting that, comparing to the other init/main agent where the `runAsUser` is implicitly defined. For this particular docker image used in the example that UID does not work, so if someone will copy/paste the example and try to run it would fail. That's the reason I wanted to address it.

Reviewer: > I thought that was the whole point of documenting that, comparing to the other init/main agent where the `runAsUser` is implicitly defined.

Honestly I wasn't sure myself 😅 Was trying to hunt that down as well as why it was failing

> For this particular docker image used in the example that UID does not work, so if someone will copy/paste the example and try to run it would fail. That's the reason I wanted to address it.

Ah so it's this particular image that fails with that UID. I was thinking it was a deeper problem 😅 . Thanks for clarifying!

You'll see the workflow complete successfully.

> **Note**: The service account running the workflow needs at least the following permissions. If <= v3.2 you must replace `workflowtasksets/status` with `workflowtasksets`.

Reviewer: I think this block should go in the `## Configuration` section above a subsection `### Permissions`. It's more important than a "Note" -- this is required

Reviewer: The Argo "agent" is mentioned in the [HTTP template docs](https://argo-workflows.readthedocs.io/en/release-3.4/http-template/#argo-agent) as well. Ideally we'd unify them and have a page on the Agent that both link to with regard to permissions / more information. The Agent is poorly documented; I didn't even know much about it until a few months ago

Potentially a section in the existing [Workflow RBAC page](https://argo-workflows.readthedocs.io/en/release-3.4/workflow-rbac/) might make sense, which is what the [quick-start manifest already links to](https://github.com/argoproj/argo-workflows/blob/2f15709952075ba208cf45ec459e66a14fa37d02/manifests/quick-start/base/agent-role.yaml#L1)

Reviewer: I would leave the "If <= v3.2" out of this and leave it in the Workflow description annotation like [the quick start manifest](https://github.com/argoproj/argo-workflows/blob/2f15709952075ba208cf45ec459e66a14fa37d02/manifests/quick-start/base/agent-role.yaml). Let's keep the two identical.

Even better would be to make that an in-line comment on the `workflowtasksets/status` line in both

Author: Let me know what you think about the latest change. I added the `### Permissions` section with some relevant links and removed the `Role` definition. Not every user may experience permission issues, because it depends on how they installed WF. I think this new section letting them know they may need to adjust the permissions are adequate for now.

I agree that the documentation for the agent and permissions, in general, could be better. Let me think about it a bit more, and for now, I'll provide this as a stopgap. I may create a follow-up PR to improve the permissions documentation, but first, I need to dig more into the code to fully understand it.

Reviewer: > I think this new section letting them know they may need to adjust the permissions are adequate for now.

Yea we'd still leave this section or something similar if we move the actual Agent docs to its own page; this section would just link to that, something like "Plugins use the [Argo Agent](workflow-rbac.md#agent), ensure you have appropriate RBAC for your Workflow".

> Let me know what you think about the latest change.

It looks like the current text is mostly just a copy of the HTTP Template doc? That section causes confusion as-is the way it is written so if we're going to add it, I'd prefer to unify in the Workflow RBAC page. We can just have a short description there and show the RBAC. Can improve that in later PRs

I can help guide that if you're ok relying on my knowledge.
